src/main.rs

```diff
@@ -827,6 +831,10 @@ fn arg_parser() -> App<'static, 'static> {
                     Arg::from_usage( "--silent='Don't print the summary profiling data every second'")
                         .required(false)
                 )
+                .arg(
+                    Arg::from_usage("--flamegraph-precision=[PRECISION] 'Minimum flame width = 1/PRECISION %'")
```

Comment:
I find this somewhat confusing to read. The "precision" abstraction is nice -- higher number gives you greater resolution in the call stacks, and lower gives you less precision. Mentally converting the precision into a width isn't intuitive, though (at least to me).

There may be a clean solution. You mentioned in the PR description that you tried to use a float type for the parameter but couldn't add it to the `Record` struct. I think we can unblock that by removing the `Eq`, `Ord`, and `Hash` traits from the struct. Some edge case bugs around float comparisons will be possible with that struct, but we only compare them in the tests (mostly to check the strings), so I think we'll be okay. If you remove those traits and switch the parameter types to `f64`, everything should work. And then rbspy will be consistent with other tools that use a "minimum width" argument like the original flamegraph library does.

Here's a diff of the key change:

```diff
--- a/src/main.rs
+++ b/src/main.rs
@@ -88,7 +88,7 @@ arg_enum!{
 }
 
 /// Subcommand.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[derive(Clone, PartialEq, PartialOrd, Debug)]
 enum SubCmd {
     /// Record `target`, writing output `output`.
     Record {
@@ -110,7 +110,7 @@ enum SubCmd {
 use SubCmd::*;
 
 /// Top level args type.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[derive(Clone, PartialEq, PartialOrd, Debug)]
 struct Args {
     cmd: SubCmd,
 }
@@ -301,7 +301,7 @@ fn snapshot(pid: Pid) -> Result<(), Error> {
 }
 
```

Reply:
Ah yes, indeed I wasn't sure what these traits were used for so I didn't even try to remove them, but you're right using a float is better here, I just applied these changes in d6bdf84fcf6548de6c5ecc3c7efff756afc780aa but what do you think about the parameter name then? I suppose "flamegraph-precision" doesn't make much sense now so do you have any recommendation? I'm thinking maybe "flame-min-width"?

Reply:
flame-min-width sounds good to me. It's reasonably short (easy on the old fingers), and its purpose should be clear enough to anyone who decides to use it.

Reply:
All right! I just pushed the rename in dbcb6a80813dd9285f463be854ff513e42ea5261. Thanks!